src/syncerd/bitcoin_syncer.rs

```diff
@@ -930,42 +930,41 @@ fn sweep_polling(
             let state_guard = state.lock().await;
             let sweep_addresses = state_guard.sweep_addresses.clone();
             drop(state_guard);
-            match Client::new(&electrum_server) {
-                Err(err) => {
-                    error!(
-                        "Failed to create btc sweep electrum client: {}, retrying",
-                        err
-                    );
-                }
-                Ok(client) => {
-                    for (id, sweep_address_task) in sweep_addresses.iter() {
-                        if let SweepAddressAddendum::Bitcoin(addendum) =
-                            sweep_address_task.addendum.clone()
-                        {
-                            let sweep_address_txs = sweep_address(
-                                addendum.source_private_key,
-                                addendum.source_address,
-                                addendum.destination_address,
-                                &client,
-                                network,
-                            )
-                            .unwrap_or_else(|err| {
-                                warn!("error polling sweep address {:?}, retrying", err);
-                                vec![]
-                            });
-                            debug!("sweep address transaction: {:?}", sweep_address_txs);
-                            if !sweep_address_txs.is_empty() {
+            if !sweep_addresses.is_empty() {
+                match Client::new(&electrum_server) {
+                    Err(err) => {
+                        error!(
+                            "Failed to create btc sweep electrum client: {}, retrying",
+                            err
+                        );
+                    }
+                    Ok(client) => {
+                        for (id, sweep_address_task) in sweep_addresses.iter() {
```

Comment:
i would put this for loop where `if !sweep_addresses.is_empty()` is, and remove the latter

Reply:
The problem is that you are then creating a new electrum client on every iteration. This was a feedback from @h4sh3d when he reviewed the sweep PR.You'd only want to create one once when there are some addresses to be swept.

Reply:
ACK